components/director/internal/open_resource_discovery/accessstrategy/access_strategy.go

```diff
@@ -0,0 +1,72 @@
+package accessstrategy
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+// UnsupportedAccessStrategyErr is an error produced when execution of unsupported access strategy takes place.
+var UnsupportedAccessStrategyErr = errors.New("unsupported access strategy")
+
+// AccessStrategy is an ORD object
+type AccessStrategy struct {
+	Type              AccessStrategyType `json:"type"`
+	CustomType        AccessStrategyType `json:"customType"`
+	CustomDescription string             `json:"customDescription"`
+}
+
+// AccessStrategyType represents the possible type of the AccessStrategy
+type AccessStrategyType string
+
+// IsSupported checks if the given AccessStrategy is supported by CMP
+func (a AccessStrategyType) IsSupported() bool {
+	_, ok := supportedAccessStrategies[a]
+	return ok
+}
+
+// Execute executes given access strategy if supported, UnsupportedAccessStrategyErr otherwise
+func (a AccessStrategyType) Execute(ctx context.Context, client *http.Client, url string) (*http.Response, error) {
```

Comment:
this should be a method on the strategy, atleast logically you dont execute the type. I understand that here you've tried to hide looking within the map of strategies and checking if executor for type exists - probably the correct flow should be ExecutorProvider.Provide(strategy string) and then Executor.Execute (rather than strategyTypes.IsSupported and then strategyType.Execute)

Reply:
But the AccessStrategy itself is just a struct with only a type field. Then in the end it is a more complicated way to execute something based on the acessStrategy's type.